torch/autograd/grad_mode.py

```diff
@@ -31,13 +32,26 @@ def _wrap_generator(self, func):
         @functools.wraps(func)
         def generator_context(*args, **kwargs):
             gen = func(*args, **kwargs)
-            while True:
-                try:
-                    with self.__class__():
-                        x = next(gen)
-                    yield x
-                except StopIteration:
-                    break
+            cls = type(self)
+            try:
+                with cls():
+                    response = gen.send(None)
```

Comment:
I think it would be nice to have a couple lines of comments in this function for future readers.
Explaining that this starts the function, and what each except below is doing.

Reply:
I will comment the code, keeping it as succinct as possible.